### PR TITLE
chore: Upgrade React Native to 0.75.1

### PR DESCRIPTION
**Pull request by tjzel**
🚀

### CODE REVIEW

File: packages/react-native-reanimated/RNReanimated.podspec [hunk 1]
```diff
@@ -23,7 +23,7 @@ def self.install_modules_dependencies_legacy(s)
   s.dependency "React-Core"
   if $new_arch_enabled
     s.dependency "React-RCTFabric"
-    s.dependency "React-Codegen"
+    s.dependency "ReactCodegen"
```
Comment (tjzel) [inline packages/react-native-reanimated/RNReanimated.podspec:26]: https://reactnative.dev/blog/2024/08/12/release-0.75#ios

File: packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NodesManager.java
Comment (tjzel) [inline packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NodesManager.java:1]: https://reactnative.dev/blog/2024/08/12/release-0.75#android

File: .github/workflows/reanimated-typescript-compatibility-test-nightly.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
     strategy:
       matrix:
         # TODO: Fetch versions from common source.
-        react-native-version: ['0.72', '0.73', '0.74', '0.75.0-rc.6', nightly]
+        react-native-version: ['0.72', '0.73', '0.74', '0.75.0', nightly]
```
Comment (szydlovsky) [inline .github/workflows/reanimated-typescript-compatibility-test-nightly.yml:19]: Why not 0.75.1 in here? Or straight up 0.75 and let it choose the latest

**Review by szydlovsky — APPROVED** [review]
G

File: .github/workflows/reanimated-typescript-compatibility-test-nightly.yml (hunk 2 above)
Comment (tjzel) [reply .github/workflows/reanimated-typescript-compatibility-test-nightly.yml:19]: You are right, it should've been 0.75

**Pull request merged**